doc/release/upcoming_changes/18737.new_feature.rst

```diff
@@ -0,0 +1,9 @@
+generic optimized BLAS and LAPACK via BLAS_LIBS and LAPACK_LIBS
```

Comment:
Can you call it `NPY_BLAS_LIBS` and `NPY_LAPACK_LIBS`?

Reply:
I'm not inclined to die on that hill, but I'd rather keep the variable names because the point is that they are _not_ specific to numpy. It is customary in other builds (probably a relativel high fraction of not widely known scientific packages or private Makefiles, even) to use BLAS_LIBS. My argument is prior use of these names for exactly this intended purpose.

Reply:
To me that's a good reason to make it more specific. Especially since it overrides the other options, this has the potential to silently change existing build workflows for people who happen to have `BLAS_LIBS` defined for some other tool.

Env vars are a terrible way of configuring builds in general, so making them numpy-specific at least provides some isolation - really everything except `CC`, `CXX`, `CFLAGS` and `CXXFLAGS` should start with `NPY_`.

Reply:
Hehe, that mirrors my argument against all this smart detection logic that by itself decides on openblas if it finds it … the build result suddenly changes depending on what variants are also installed in the system. But as I said, I won't fight that point. I'll change it to NPY_ prefix.